libplatsupport/src/plat/imx6/clock.c

```diff
@@ -71,15 +71,16 @@
 #define CLKGATE_MODE_ON_ALL     0x3
 #define CLKGATE_MODE_MASK       0x3
 
-#define CLKO1_SRC_AHB       (0xBU << 0)
-#define CLKO1_SRC_IPG       (0xCU << 0)
-#define CLKO1_SRC_MASK      (0xFU << 0)
-#define CLKO1_ENABLE        (1U << 7)
-
-#define CLKO2_SRC_MMDC_CH0  (0U << 21)
-#define CLKO2_SRC_MASK      (0x1FU << 16)
-#define CLKO2_ENABLE        (1U << 24)
-#define CLKO_SEL            (1U << 8)
+/* CCM Clock Output Source (CCM_CCOSR) */
+#define CLKO1_SRC_AHB       0xBU
+#define CLKO1_SRC_IPG       0xCU
+#define CLKO1_SRC_MASK      0xFU /* bit 0-3 */
+#define CLKO1_ENABLE        BIT(7)
+#define CLKO_SEL            BIT(8) /* select CCM_CLKO1 or CCM_CLKO2 */
+#define CLKO2_SRC_MMDC_CH0  (0U << 16)
```

Comment:
Can you confirm that this should be `(0U << 16)` and not the previous `(0U << 21)`?

Reply:
Yes. The TRM says that bit 16-20 is `CLKO2_SEL` and the value `mmdc_ch0_clk_root` is b00000. Bit 21-23 is `CLKO2_DIV`. The function `_clko_init()` will set this value:
```
            v &= ~CLKO2_SRC_MASK;
            v |= CLKO2_SRC_MMDC_CH0;
```
The old value seem wrong. But it has worked because it is all zeros anyway. Anyway, I've made a dedicated commit "libplatsupport/imx6: fix CLKO2_SRC_MMDC_CH0" for this now.